test/run-e2e-tests.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+###############################################################################
+# Copyright Contributors to the Open Cluster Management project
+###############################################################################
+
+set -o errexit
+set -o nounset
+
+function wait_deployment() {
+  set +e
+  for((i=0;i<30;i++));
+  do
+    ${KUBECTL} -n $1 get deploy $2
+    if [ 0 -eq $? ]; then
+      break
+    fi
+    echo "sleep 1 second to wait deployment $1/$2 to exist: $i"
+    sleep 1
+  done
+  set -e
+}
+
+function hub_approve_cluster(){
+  local cluster_name=$1
+  echo "approve ${cluster_name}"
+  ${KUBECTL} get csr -l open-cluster-management.io/cluster-name=${cluster_name} | \
+    grep -v NAME | awk '{print $1}' | xargs ${KUBECTL} certificate approve
+  ${KUBECTL} patch managedcluster "${cluster_name}" -p='{"spec":{"hubAcceptsClient":true}}' --type=merge
+  ${KUBECTL} get managedcluster "${cluster_name}"
+}
+
+BUILD_DIR="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+REPO_DIR="$(dirname "$BUILD_DIR")"
+WORK_DIR="${REPO_DIR}/_output"
+
+KIND_VERSION="v0.11.1"
+KIND="${WORK_DIR}/bin/kind"
+
+KUBE_VERSION="v1.20.2"
+KUBECTL="${WORK_DIR}/bin/kubectl"
+
+export MANAGED_CLUSTER_NAME="cluster1"
```

Comment:
i would call it hub and managedcluster, you can run a klusterlet on managed cluster and a klusterlet on hub which manages managed cluster.